doc/api/url.md

```diff
@@ -230,11 +230,28 @@ manner similar to that of a Web browser resolving an anchor tag HREF.
 For example:
 
 ```js
-url.resolve('/one/two/three', 'four')         // '/one/two/four'
-url.resolve('http://example.com/', '/one')    // 'http://example.com/one'
-url.resolve('http://example.com/one', '/two') // 'http://example.com/two'
+url.resolve('/one/two/three', 'four');         // '/one/two/four'
+url.resolve('http://example.com/', '/one');    // 'http://example.com/one'
+url.resolve('http://example.com/one', '/two'); // 'http://example.com/two'
 ```
 
+Some protocols (`http`, `https`, `ftp`, `gopher` and `file`) are treated as special cases.
+Every other (for instance ftps or wss) is treated differently whether its URL 
+ends in a slash or not.
+
+For example:
+
+```js
+url.resolve('https://foo.tld', 'bar');   // 'https://foo.tld/bar'
+url.resolve('wss://foo.tld', 'bar');     // 'wss://bar'
+url.resolve('ftps://foo.tld', 'bar');    // 'ftps://bar'
+```
+
+Although, the uniform behaviour of the method is achieved when the URL ends with a slash.
```

Comment:
Sorry, that isn't helpful - the point of the docs is to explain, and they aren't, so having me go read a node issue to understand a doc change is a bit backwards.

As is, this PR adds some extra examples and some lines of text that basically say "look carefully at the examples to see how this API works, paying particular attention to the protocol scheme, because it changes the behaviour". This is arguably an improvement over the existing doc, if @addaleax is happy with that, I've no objection.

However, it still doesn't document the API behaviour. If you have the interest, the existing docs could be improved even more. If not, I understand.